web3/types.py

```diff
@@ -81,6 +82,9 @@
     "blockNumber": int,
 })
 
+JsonRpcResponse = Dict[str, Any]
+
+Middleware = Callable[[Any, Any], Any]
```

Comment:
I think that these `Any` types can be tightened some, at least for the arguments to the middleware callables.